core/src/main/java/org/opensearch/remote/metadata/client/BulkDataObjectResponse.java

```diff
@@ -88,11 +139,33 @@ public boolean hasFailures() {
         return this.failures;
     }
 
+    /**
+     * Returns the BulkhResponse object
+     * @return the bulk response if present, or parsed otherwise
+     */
+    public @Nullable BulkResponse bulkResponse() {
+        if (this.bulkResponse == null) {
+            try {
+                return BulkResponse.fromXContent(parser);
+            } catch (IOException | NullPointerException e) {
+                return null;
+            }
+        }
+        return this.bulkResponse;
+    }
+
     /**
      * Returns the parser
      * @return the parser
      */
     public XContentParser parser() {
+        if (this.parser == null) {
+            try {
+                return SdkClientUtils.createParser(bulkResponse);
+            } catch (IOException | NullPointerException e) {
+                return null;
```

Comment:
What is the cascade effect if we just return null to the downstream?

Reply:
This is one of those “should never happen” situations. 

The SDKClientUtil null checks it if it’s used.  

Otherwise the caller has to null check.

Reply:
Alternative is throwing an unchecked exception the caller doesn’t know to catch, or a checked exception that will never happen that we force the caller to handle